cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/SortedSSTableWriter.java

```diff
@@ -137,8 +141,24 @@ public void addRow(BigInteger token, Map<String, Object> boundValues) throws IOE
         }
         // rows are sorted. Therefore, only update the maxToken
         maxToken = token;
-        cqlSSTableWriter.addRow(boundValues);
-        rowCount += 1;
+        try
+        {
+            cqlSSTableWriter.addRow(boundValues);
+            rowCount += 1;
+        }
+        catch (Throwable t)
+        {
+            if (t.getCause() != null && t.getCause().getClass().getName().equals("org.apache.cassandra.cql3.constraints.ConstraintViolationException"))
```

Comment:
This is questionable but ... it is here to get the idea. What I am not sure about is that if we moved it "closer" to the source, cqlSSTableWriter is basically a bridge to real implementation. So if we moved it to a bridge, we would need to copy it all over again for every bridge there is / will be.

For now we have only Cassandra 4 bridge and I can not put it there anyway because the concept of "constraints" was not known back in Cassandra 4, it will be there for the first time in Cassandra 5.1 / 6.0 we do not have any bridge for yet.

Reply:
I was going to suggest using `ThrowableUtils` , but it looks like the Analytics version of this class is less featured than the Sidecar version of `ThrowableUtils` .

Reply:
Tiny NIT:
```suggestion
            if (t.getCause() != null && "org.apache.cassandra.cql3.constraints.ConstraintViolationException".equals(t.getCause().getClass().getName()))
```